src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -146,9 +152,20 @@ public void ICollection_NonGeneric_SyncRoot_NonNull(int count)
 
         [Theory]
         [MemberData(nameof(ValidCollectionSizes))]
+        public void ICollection_NonGeneric_SyncRoot_Null(int count)
+        {
+            if (ICollection_NonGeneric_SupportsSyncRoot && ICollection_NonGeneric_HasNullSyncRoot)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                Assert.Null(collection.SyncRoot);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
         public void ICollection_NonGeneric_SyncRootConsistent(int count)
         {
-            if (ICollection_NonGeneric_SupportsSyncRoot)
+            if (ICollection_NonGeneric_SupportsSyncRoot && !ICollection_NonGeneric_HasNullSyncRoot)
```

Comment:
Is this change needed?  Looks like the test will work even if the collection returns null from SyncRoot.  If we're going to run the test anyway, we may as well do something with it.